apps/condo/domains/meter/schema/RegisterMetersReadingsService.js

```diff
@@ -0,0 +1,266 @@
+                '}',
+        },
+        {
+            access: true,
+            type: 'input RegisterMetersReadingsReadingAddressMetaInput { unitType: String!, unitName: String!, globalId: String }',
+        },
+        {
+            access: true,
+            type: 'input RegisterMetersReadingsReadingInput {' +
+                'address: String!,' +
+                'addressMeta: RegisterMetersReadingsReadingAddressMetaInput!,' +
+                'accountNumber: String!,' +
+                'meterNumber: String!,' +
+                'meterResource: MeterResourceWhereUniqueInput!,' +
+                'date: String!,' +
+                'value1: String!,' +
+                'value2: String,' +
+                'value3: String,' +
+                'value4: String,' +
+                'meterMeta: RegisterMetersReadingsMeterMetaInput' +
+                '}',
+        },
+        {
+            access: true,
+            type: 'input RegisterMetersReadingsInput { dv: Int!, sender: SenderFieldInput!, organization: OrganizationWhereUniqueInput!, readings: [RegisterMetersReadingsReadingInput!]! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canRegisterMetersReadings,
+            schema: 'registerMetersReadings(data: RegisterMetersReadingsInput!): [MeterReading]',
+            resolver: async (parent, /**{ data: RegisterMetersReadingsInput }*/args, context) => {
+                const { data: { dv, sender, organization, readings } } = args
+
+                if (readings.length > READINGS_LIMIT) {
+                    throw new GQLError(ERRORS.TOO_MUCH_READINGS(readings.length), context)
+                }
+
+                /** @type Organization */
+                const organizationData = await getById('Organization', organization.id)
+
+                if (!organizationData || !!organizationData.deletedAt) {
+                    throw new GQLError(ERRORS.ORGANIZATION_NOT_FOUND, context)
+                }
+
+                const locale = extractReqLocale(context.req) || conf.DEFAULT_LOCALE
+
+                const propertyResolver = new PropertyResolver({ context })
+                propertyResolver.tin = organizationData.tin
+
+                const resolvedAddresses = await propertyResolver.normalizeAddresses(readings.reduce((res, reading, index) => ({
+                    ...res,
+                    [reading.address]: {
+                        address: reading.address,
+                        addressMeta: reading.addressMeta,
+                    },
+                }), {}))
+
+                const addressesKeys = readings.map((reading) => get(resolvedAddresses, [reading.address, 'addressResolve', 'propertyAddress', 'addressKey'])).filter(Boolean)
+
+                /** @type Property[] */
+                const properties = await find('Property', {
+                    organization,
+                    deletedAt: null,
+                    addressKey_in: addressesKeys,
+                })
```

Comment:
Maybe we should add an index for `addressKey` field?
And use `loadListByChunks` so that there is no dependence on the readings limit? :)

Reply:
> Maybe we should add an index for `addressKey` field?

It's better to move it to another task if really needed

> And use loadListByChunks so that there is no dependence on the readings limit?

I do not need all relations here